src/core/utils/route-builder.ts

```
@@ -102,74 +109,61 @@ export function createWalkingSegment(
+    id: routeId,
+    segments,
     totalDuration,
-    transfers: transferCount,
-    id,
+    totalDistance,
+    transfers,
+    totalStops,
   };
+
+  // If a source route was provided with a requestedOrigin, preserve it
+  if (sourceRoute && sourceRoute.requestedOrigin) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)